Core/Src/can_handler.c

```diff
@@ -108,31 +108,57 @@ void vCanDispatch(void *pv_params)
 	HAL_StatusTypeDef msg_status;
 
 	for (;;) {
+		osThreadFlagsWait(CAN_DISPATCH_FLAG, osFlagsWaitAny,
+				  osWaitForever);
 		/* Send CAN message */
-		if (osOK == osMessageQueueGet(can_outbound_queue,
-					      &msg_from_queue, NULL,
-					      osWaitForever)) {
+		while (osMessageQueueGet(can_outbound_queue, &msg_from_queue,
+					 NULL, 0U) == osOK) {
 			msg_status = can_send_msg(can1, &msg_from_queue);
+
 			if (msg_status == HAL_ERROR) {
 				fault_data.diag = "Failed to send CAN message";
 				queue_fault(&fault_data);
 			} else if (msg_status == HAL_BUSY) {
 				fault_data.diag = "Outbound mailbox full!";
 				queue_fault(&fault_data);
-			} else {
-				//printf("Message sent: %lX\r\n", msg_from_queue.id);
 			}
 		}
-
-		osDelay(CAN_DISPATCH_DELAY);
 	}
 }
 
-int8_t queue_can_msg(can_msg_t msg)
+osThreadId_t can_receive_thread;
+const osThreadAttr_t can_receive_attributes = {
+	.name = "CanProcessing",
+	.stack_size = 128 * 8,
+	.priority = (osPriority_t)osPriorityRealtime,
+};
+
+void vCanReceive(void *pv_params)
 {
-	if (!can_outbound_queue)
-		return -1;
+	dti_t *mc = (dti_t *)pv_params;
 
-	osMessageQueuePut(can_outbound_queue, &msg, 0U, 0U);
-	return 0;
-}
+	can_msg_t msg;
+
+	for (;;) {
+		osThreadFlagsWait(NEW_CAN_MSG_FLAG, osFlagsWaitAny,
+				  osWaitForever);
+		while (osOK ==
+		       osMessageQueueGet(can_inbound_queue, &msg, 0U, 0U)) {
+			switch (msg.id) {
+			/* Messages Relevant to Motor Controller */
+			case DTI_CANID_ERPM:
+				dti_record_rpm(mc, msg);
+				break;
+				// case DTI_CANID_CURRENTS:
```

Comment:
Imo we should just remove unused dti messages and remove them from the filter list above.  I dont think we will ever need to be fetching anything more than DTI RPM, and if we do we can add it back.

Reply:
yeah, simplification here is good. Any messages that we unliekly to care about, leave them for the future, we probably will never want them so no need to accomdate now